### PR TITLE
[Refactor] Simplify column predicates using template

### PR DESCRIPTION
**Pull request by dirtysalt**
To remove duplicated code.

### CODE REVIEW

File: be/src/storage/vectorized/column_eq_predicate.cpp [hunk 1]
```diff
@@ -27,52 +27,33 @@ class ColumnEqPredicate : public ColumnPredicate {
 
     ~ColumnEqPredicate() override = default;
 
-    void evaluate(const Column* column, uint8_t* selection, uint16_t from, uint16_t to) const override {
+    template <typename Op>
+    inline void t_evaluate(const Column* column, uint8_t* selection, uint16_t from, uint16_t to) const {
```
Comment (kangkaisen) [inline be/src/storage/vectorized/column_eq_predicate.cpp:31]: What's the meaning of `t` ?

Comment (dirtysalt) [reply be/src/storage/vectorized/column_eq_predicate.cpp:31]: `template` :)

File: be/src/storage/vectorized/column_ge_predicate.cpp [hunk 2]
```diff
@@ -125,36 +106,33 @@ class BinaryColumnGePredicate : public ColumnPredicate {
...
             /* must use const uint8_t* to make vectorized effect, vector<uint8_t> not work */
             const uint8_t* is_null = down_cast<const NullableColumn*>(column)->immutable_null_column_data().data();
             for (size_t i = from; i < to; i++) {
-                sel[i] = (!is_null[i]) && (v[i] >= _value);
+                sel[i] = Op::apply(sel[i], (uint8_t)((!is_null[i]) && (v[i] >= _value)));
             }
         }
     }
 
+    void evaluate(const Column* column, uint8_t* selection, uint16_t from, uint16_t to) const override {
```
Comment (darionyaphet) [inline be/src/storage/vectorized/column_ge_predicate.cpp:126]: This part looks a little generic.

**Review by kangkaisen — DISMISSED** [review]

File: be/src/storage/vectorized/column_ge_predicate.cpp (hunk 2 above)
Comment (dirtysalt) [reply be/src/storage/vectorized/column_ge_predicate.cpp:126]: Indeed.

**Review by kangkaisen — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Pull request merged**